### ISSUE TITLE
Makefile for Darwin

### ISSUE THREAD

**Issue by kuba--** [tinycthread/tinycthread#19]
For Darwin (macosx) you do not have to link librt, so I suggest to modify part of the Makefile:
```Makefile
# Non-windows systems need pthread and rt (for Linux)
ifndef WINDOWS
LIBS += -lpthread
ifeq ($(shell uname -s), Linux)
LIBS += -lrt
endif
endif
```

**Comment by Leandros** [comment]
`uname -s` outputs `Darwin` on OS X. But there is an issue with `uname -s`, because it's not available on cygwin's `uname` command. The punchline is, simply `uname` also returns `Darwin`.

The OS detection I normally use in my makefiles:

```
OS =
ifdef SystemRoot
    OS = win
else
    ifeq ($(shell uname), Linux)
        OS = linux
        RM = rm -f
        FixPath = $1
    endif
    ifeq ($(shell uname), Darwin)
        OS = osx
    endif
    ifeq ($(shell uname), CYGWIN_NT-6.1)
        OS = cygwin
    endif
endif
```

**Comment by nemequ** [comment]
Fixed in e93b3e33c59ab89e371e29c4b6f7db444d643c44
